src/Http/Http/src/Internal/DefaultHttpResponse.cs

```diff
@@ -70,8 +66,13 @@ public override IHeaderDictionary Headers
 
         public override Stream Body
         {
-            get { return HttpResponseFeature.Body; }
-            set { HttpResponseFeature.Body = value; }
+            get { return HttpResponseBodyFeature.Body; }
+            set
+            {
+                var feature = new StreamResponseBodyFeature(value);
```

Comment:
I had an idea. We could tell people to implement `IHttpResponseBodyFeature` on their `Stream` and we'll detect that and use the feature when you set the body.

Reply:
People implementing IHttpResponseBodyFeature know enough to call the feature collection directly. People calling the body setter are doing it with a naked stream like MemoryStream.

Reply:
Fair enough.